bentoml/cli/__init__.py

```diff
@@ -259,31 +280,53 @@ def serve(port, bento=None, with_conda=False):
         default=False,
         help="Run API server in a BentoML managed Conda environment",
     )
-    def serve_gunicorn(port, workers, timeout, bento=None, with_conda=False):
+    @click.option(
+        '--enable-marshal',
+        is_flag=True,
+        default=False,
+        help="(Alpha) Run API server with marshal service",
+    )
+    def serve_gunicorn(port, workers, timeout, bento=None, with_conda=False,
+                       enable_marshal=False):
         track_cli('serve_gunicorn')
         bento_service_bundle_path = resolve_bundle_path(
             bento, pip_installed_bundle_path
         )
 
         if with_conda:
-            run_with_conda_env(
-                pip_installed_bundle_path,
-                'bentoml serve_gunicorn {bento} -p {port} -w {workers} '
-                '--timeout {timeout}'.format(
-                    bento=bento_service_bundle_path,
-                    port=port,
-                    workers=workers,
-                    timeout=timeout,
-                ),
-            )
-            return
-
-        from bentoml.server.gunicorn_server import GunicornBentoServer
+            def _run_api_server(_api_server_port):
+                run_with_conda_env(
+                    pip_installed_bundle_path,
+                    'bentoml serve_gunicorn {bento} -p {port} -w {workers} '
```

Comment:
consider just add `--enable-marshal` here in the cli arg? that way the marshal server will also be running in a conda environment and probably will simplify the code here